### PR TITLE
Recepis App: Create Migrations and Schema

### PR DESCRIPTION
**Pull request by luccamicale**
## Create Migrations and Schema 💯 

##https://github.com/AlaaShaheen2/Recipe-app 👍 

## Checklist before requesting a review
- [ ] we have performed a self-review of my code
- [ ] we create the migrations 
- [ ] we create de schema using rails db:migrate 👍 🥇 🔢 💯

### CODE REVIEW

File: db/migrate/20230213211102_create_foods.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateFoods < ActiveRecord::Migration[7.0]
+  def change
+    create_table :foods do |t|
+      t.string :name
+      t.string :measurent_unit
+      t.decimal :price
+      t.string :quantity
+      t.integer :user_id
```
Comment (AlaaShaheen2) [inline db/migrate/20230213211102_create_foods.rb:8]: I am deleting this column (user_id) using a new migration and will add it as a foreign key.

File: db/migrate/20230213212021_create_recipes.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateRecipes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :recipes do |t|
+      t.string :name
+      t.string :preparation_time
```
Comment (AlaaShaheen2) [inline db/migrate/20230213212021_create_recipes.rb:5]: datatype for this column should be an integer (I am updating it).

File: db/migrate/20230213212021_create_recipes.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      t.string :cooking_time
```
Comment (AlaaShaheen2) [inline db/migrate/20230213212021_create_recipes.rb:6]: Cooking time is also integer not string

File: db/migrate/20230213212021_create_recipes.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      t.text :description
+      t.boolean :public
+      t.integer :user_id
```
Comment (AlaaShaheen2) [inline db/migrate/20230213212021_create_recipes.rb:9]: This is not a column but a foreign key. I am removing this and will add it as a foreign key.

File: db/migrate/20230213211430_create_recipe_foods.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateRecipeFoods < ActiveRecord::Migration[7.0]
+  def change
+    create_table :recipe_foods do |t|
+      t.string :quantity
+      t.integer :recipe_id
```
Comment (AlaaShaheen2) [inline db/migrate/20230213211430_create_recipe_foods.rb:5]: Same here not a foreign key.

File: db/migrate/20230213211430_create_recipe_foods.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+      t.integer :food_id
```
Comment (AlaaShaheen2) [inline db/migrate/20230213211430_create_recipe_foods.rb:6]: And same here, not a foreign key.

**Review by AlaaShaheen2 — APPROVED** [review]
Hi @luccamicale,

You made a good effort in the project.

I reviewed the code and found some issues, but I fixed everything and commented for your reference.

Just take the comments as a reference for the upcoming milestones.

I did the followings:

- Edited Data types for columns which were supposed to be integers instead of string.
- Removed the columns which were supposed to be foreign keys.
- Added foreign keys.

**Comment by luccamicale** [comment]
Hi @AlaaShaheen2 Thanks for your review.  Now I will go the merge branch with development
💯 👍 🔢 🥇

**Pull request merged**